Fix adr path and fetching of user email

KLBa opened the pull request:
This PR attempts to fix #149. For this, two things are changed:

* The `adrPath` attribute of a repository is set when loading the repository in `api.js` → `loadARepositoryContent()`. Before it was only read in `store.js` → `createNewAdr()` but never written.
* The endpoint to get the E-Mail is changed from `/user/emails` to `/user/public_emails`. The mail is required for commits.
  (Docs for the endpoints here: <https://docs.github.com/en/rest/reference/users#list-public-email-addresses-for-the-authenticated-user>.)
  I'm not sure why `/user/emails` did not work; maybe the OAuth token scope `user:email` isn't needed for `/user/public_emails` despite the docs?

  Also, the mail doesn't have to be the primary one, e.g., the `@users.no-reply.github.com` mails are not primary. However, if a primary mail exists, it is preferred. (See `store.js` → `setEmail()` for the implementation)

Sadly, testing is kinda hard. Maybe Cypress could mock the REST API, but I didn't look into it.
I only updated the test in `PushNewAdr.js` to check if the displayed file name in the commit dialog is not empty.

KLBa commented:
Looks like the cypress tests use the ADR-Manager repository for testing, and the `cy.intercept`s do nothing.
This is sub-optimal, because a working commit functionality will spam the `main` branch with test commits ...

xJREB commented:
> I removed (force-pushed) the test commits from the main branch, but for reference e.g., this commit https://github.com/adr/adr-manager/commit/dc428327f30a0a76bdfdc9037078e067e1b6c21e was created by the GitHub Action and not by @xJREB.

Yeah, I guess, you're correct, @KLBa, thank you. I think creating a test repo under the ADR organization would probably be the way to go.

koppor commented:
I created the repository https://github.com/adr/adr-manager-alpha. A push to the branch `publish` triggers a bulid and a deploy on gh-pages. The repository is public to enable GitHub pages for free. 

The current state is available at https://adr.github.io/adr-manager-alpha/#/ - and ready to test!

KLBa commented:
@koppor  I added a `testing-branch` to <https://github.com/adr/adr-manager-alpha> (to avoid that the commits show up as public contributions on user profiles...)
I also updated the E2E test to use that branch. [My testing already spammed it with some commits](https://github.com/adr/adr-manager-alpha/commits/testing-branch).

The GitHub Action still impersonates @xJREB though, idk if that's a good idea :)

xJREB commented:
> @koppor I added a `testing-branch` to https://github.com/adr/adr-manager-alpha (to avoid that the commits show up as public contributions on user profiles...) I also updated the E2E test to use that branch. [My testing already spammed it with some commits](https://github.com/adr/adr-manager-alpha/commits/testing-branch).

Thank you! I don't know if a full-blown copy of the ADR Manager repo would have been necessary, since we only needed a repo to commit and push ADRs. But maybe I'm also missing something. :)

> The GitHub Action still impersonates @xJREB though, idk if that's a good idea :)

Hmm, true, but I'm not quite sure what a much better solution would be. We could obviously create a new account for this that then gets access to `adr/adr-manager-alpha`. This would probably be a bit more secure, but I'm not terribly worried that the token gets compromised. Any other options or reasons to change to a different user that I don't see?

koppor commented:
Looks good, thus I merged to keep things going.

The only issue I have is that a blank repository, it does not work:

Steps:

1. Create an empty Repository on github

![grafik](https://user-images.githubusercontent.com/1366654/161131087-f2ccd8dd-da7b-4e4d-9fba-2d3735542ae8.png)

2. Connect to the repository

![grafik](https://user-images.githubusercontent.com/1366654/161133499-13c925cb-a3b2-458c-809a-c89909989994.png)
